src/azure-cli/azure/cli/command_modules/eventgrid/commands.py

```diff
@@ -28,23 +51,77 @@ def load_command_table(self, _):
+        client_factory=system_topics_factory,
+        client_arg_name='self'
+    )
+
+    system_topic_event_subscriptions_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#SystemTopicEventSubscriptionsOperations.{}',
+        client_factory=system_topic_event_subscriptions_factory,
+        client_arg_name='self'
+    )
+
+    partner_registrations_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#PartnerRegistrationsOperations.{}',
+        client_factory=partner_registrations_factory,
+        client_arg_name='self'
+    )
+
+    partner_namespaces_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#PartnerNamespacesOperations.{}',
+        client_factory=partner_namespaces_factory,
+        client_arg_name='self'
+    )
+
+    event_channels_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#EventChannelsOperations.{}',
+        client_factory=event_channels_factory,
+        client_arg_name='self'
+    )
+
+    partner_topics_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#PartnerTopicsOperations.{}',
+        client_factory=partner_topics_factory,
+        client_arg_name='self'
+    )
+
+    partner_topic_event_subscriptions_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#PartnerTopicEventSubscriptionsOperations.{}',
+        client_factory=partner_topic_event_subscriptions_factory,
+        client_arg_name='self'
+    )
+
     topic_type_mgmt_util = CliCommandType(
         operations_tmpl='azure.mgmt.eventgrid.operations#TopicTypesOperations.{}',
         client_factory=topic_types_factory,
         client_arg_name='self'
     )
 
+    private_endpoint_connections_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations.#PrivateEndpointConnectionsOperations.{}',
+        client_factory=private_endpoint_connections_factory,
+        client_arg_name='self'
+    )
+
+    private_link_resources_mgmt_util = CliCommandType(
+        operations_tmpl='azure.mgmt.eventgrid.operations#PrivateLinkResourcesOperations.{}',
+        client_factory=private_link_resources_factory,
+        client_arg_name='self'
+    )
+
     with self.command_group('eventgrid topic', topics_mgmt_util, client_factory=topics_factory) as g:
         g.show_command('show', 'get')
         g.command('key list', 'list_shared_access_keys')
         g.command('key regenerate', 'regenerate_key')
         g.command('delete', 'delete')
         g.custom_command('list', 'cli_topic_list')
         g.custom_command('create', 'cli_topic_create_or_update')
-        g.generic_update_command('update',
-                                 getter_name='get',
-                                 setter_name='update',
-                                 client_factory=topics_factory)
+        g.custom_command('update', 'cli_topic_update')
```

Comment:
Why do we remove `generic_update_command`? It might be breaking change for customers.

Reply:
There is not breaking change here as the parameters and commands are still the same and our tests is running in the same fashion. This gives us more control to have common methods between topic and domain and less duplicat code.

Reply:
if I am missing something, please elaborate on why you think this is breaking change.. From our perspective, the command name and parameters lists are the same... we are not breaking anything to user.

PLease share ur msft alias to discuss further if needed on why you think this might be a breaking change because we don't think it is.. our tests are passing .. and there is no CLI breaks. plz LMK if we r missing anything.. Otherwise, please approve this PR or provide a next step.

thanks

Reply:
`generic_update` would expose extra four global arguments for customers.

Reply:
We cannot reuse generic_update as we have additional parameters we need for the update, including SKU, idenity, inbound_ip_address,  etc.. these needs to be specified separately as part of the command. Keep using generic_update will result in this error: ERROR    cli.azure.cli.core.parser:parser.py:153 az eventgrid domain update: error: the following arguments are required: --domain-update-parameters

WE did have this fix in CLI extension for many months now and we did not have an issue from customer of from CLI reviewers. We push back on this comment as Won't fix as there is no user impact and any UX changes needs to be handled by proper internal team. We should not impact customer experience for this.